File: web-api/src/main/java/cn/labzen/web/api/annotation/Caching.java
package cn.labzen.web.api.annotation;

import java.lang.annotation.*;

/**
 * API 响应缓存注解。
 * <p>
 * 注解在 Controller 接口或方法上，对 API 响应内容进行缓存处理。
 * <p>
 * <b>待实现：</b> 定义缓存参数，如缓存时长、缓存策略等。
 */
@Documented
@Retention(RetentionPolicy.SOURCE)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface Caching {
}

File: src/main/kotlin/cn/labzen/web/annotation/Caching.kt
package cn.labzen.web.annotation

/**
 * 注解在 Controller 接口或方法上，对API响应内容进行缓存
 *
 * todo 待实现，定义缓存参数
 */
@MustBeDocumented
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.SOURCE)
annotation class Caching()
